### PR TITLE
feat(controller): nested & array param coercion (#16)

### LINKED ISSUE #16: No nested/array param type — can't collect Rails nested attributes (forces a per-row architecture)
## Problem

`coerce_params` only handles scalar types (`:string`, `:integer`, `:float`, `:boolean`). There's no way to declare a **nested or array** param, so a reactive action can't receive Rails-style `model[nested_attributes][i][field]` or array params (`ids[]`).

## Concrete repro (cosmos)

The invoice edit form is **one** form where flat invoice fields *and* line items (`fields_for :invoice_items` → `invoice[invoice_items_attributes][i][...]`) submit together. To make editing reactive, a single `save` action would need to accept `invoice_items_attributes` — which isn't expressible. As a result, **reactive save and reactive line-item editing become the same problem**, and the only viable design is per-row reactive components that each save independently (mirroring a separate items controller). That's a real architecture forced by the missing param type.

## Proposed API

Allow a nested schema and/or an array type, e.g.:

```ruby
action :save, params: {
  date: :string,
  invoice_items_attributes: [ { id: :integer, quantity: :float, price: :float, _destroy: :boolean } ],
  bank_account_ids: [:integer],
}
```

`coerce_params` would recurse for hash-schemas and map over arrays. This would let a reactive form mirror a normal Rails nested-attributes update in one action, instead of forcing a per-row decomposition.

(Per-row reactivity is often nicer anyway — but it should be a choice, not the only option. Found building reactive invoice line items.)

### PR DESCRIPTION
**Pull request by mhenrixon**
## Summary

`coerce_params` only handled scalar types (`:string` / `:integer` / `:float` / `:boolean`), so a reactive action couldn't accept Rails-style nested attributes (`invoice[invoice_items_attributes][i][...]`) or array params (`ids[]`). As the issue notes, that forced reactive invoice editing into a per-row component architecture rather than letting it be a choice.

## The API

A declared type is now one of:

- a **scalar** symbol — `:string` (default), `:integer`, `:float`, `:boolean`
- a **nested hash** schema — `{ id: :integer, … }`
- a **one-element array** wrapping either — `[:integer]` (array of scalar) or `[{ … }]` (array of hash)

```ruby
action :save, params: {
  date: :string,
  bank_account_ids: [:integer],
  invoice_items_attributes: [
    { id: :integer, quantity: :float, price: :float, _destroy: :boolean }
  ]
}

def save(date:, bank_account_ids:, invoice_items_attributes:)
  @invoice.update!(date:, bank_account_ids:, invoice_items_attributes:)
end
```

Coercion recurses: `coerce_hash` keeps only declared keys (no mass assignment), `coerce_array` maps the element type over the values. Array params accept **both** a real JSON array and a Rails-style index hash (`{ "0" => …, "1" => … }`), so a `fields_for` collection works either way. Scalar coercion is unchanged (the existing path was refactored into `coerce_scalar` and stayed green).

## Test plan

| Layer | What it proves |
|-------|----------------|
| `spec/requests/nested_params_spec.rb` | array-of-scalar, array-of-hash with per-field types, dropped undeclared nested keys, Rails index-hash input, omitted key stays absent, empty array preserved, flat scalars alongside nested |
| `spec/system/nested_params_spec.rb` | end-to-end browser round trip: explicit nested params posted, coerced server-side, component re-renders with the typed structure, no full reload |

New dummy `NestedParamsComponent` + `/nested_params` route (reflects the coerced structure as JSON for exact-type assertions).

## Verification

- `bundle exec standardrb` — clean
- `bundle exec rspec` — **83 examples, 0 failures**
- Existing scalar coercion unaffected

## Docs

- `lib/phlex/reactive/component.rb` — `action` DSL doc now covers array/nested types.
- `README.md` — "Array & nested params" section under the Component API.
- `CHANGELOG.md` — `Unreleased → Added`.

Closes #16

## Summary by CodeRabbit

* **New Features**
  * Enhanced reactive action parameter coercion for scalar arrays and nested hash/array (Rails-style nested attributes) schemas, coercing values per the declared types.
  * Array-shaped inputs are accepted as either JSON arrays or Rails index-hash arrays, with undeclared nested keys dropped.
* **Bug Fixes**
  * Malformed array payloads (and present-but-non-array values) are ignored to prevent unintended “clear all” behavior, while valid empty arrays still pass through.
* **Documentation**
  * Expanded API reference and changelog with schema examples and supported input shapes.
* **Tests**
  * Added request and end-to-end coverage for coercion correctness and edge cases.

### CODE REVIEW

File: app/controllers/phlex/reactive/actions_controller.rb [hunk 1]
```diff
@@ -123,18 +123,46 @@ def reactive_action_name
...
+        elsif type.is_a?(Hash)
+          coerce_hash(value, type)
+        else
+          coerce_scalar(value, type)
+        end
+      end
+
+      def coerce_array(value, element_type)
+        array_values(value).map { |element| coerce(element, element_type) }
+      end
```
[the remark this answers is not in the document]
Comment (mhenrixon) [reply app/controllers/phlex/reactive/actions_controller.rb:165]: Good catch — fixed in a02c8da.

I took a slightly different route than `raise BadRequest`: a malformed (present-but-non-array) value now returns an internal `DROP` sentinel, and `coerce_hash` skips that key — so the action falls back to its keyword default, exactly as if the client had omitted the param. This matches the rest of the coercion layer's "drop, don't trust" philosophy (undeclared keys are already silently dropped), and avoids turning a UI bug into a hard 400. Critically, it no longer reads as an explicit "clear everything" on `update!(bank_account_ids:)`. A real empty array still passes through as `[]`.

Covered by two new request specs: a scalar sent for `bank_account_ids: [:integer]` and a scalar sent for the array-of-hash param are both dropped (param absent → nil), distinct from the existing empty-array → `[]` test.
